simulation/sub8_simulation/sub8_sim_tools/rendering/world.py

```diff
@@ -100,18 +103,20 @@ def make_buffer(self, mesh):
 
 
 class Sphere(Entity):
-    _vertex_shader = Shaders.lighting['phong']['vertex']
-    _fragment_shader = Shaders.lighting['phong']['fragment']
+    _vertex_shader = Shaders.base_shaders['phong']['vertex']
+    _fragment_shader = Shaders.base_shaders['phong']['fragment']
 
-    def __init__(self, position, radius, color, **kwargs):
+    def __init__(self, position, radius, color, shader_manager=None, **kwargs):
         sphere_mesh = geometry.create_sphere(
             rows=int(radius * 30),
             cols=int(radius * 30),
             radius=radius,
             **kwargs
         )
+
         sphere_buffer, faces = self.make_buffer(sphere_mesh)
         super(self.__class__, self).__init__(sphere_buffer, faces=faces, position=position, color=color)
+        shader_manager.register_lighting_shader(self)
```

Comment:
Oh I see. My thinking is tha tit would be unusual for an Entity to change its own shader, but pretty normal for the world to change every Entity's shader at once when switching from camera to sonar rendering.

Not a breaking issue, just a thought